lib/modules/platform/github/index.ts

```diff
@@ -671,37 +593,86 @@ function matchesState(state: string, desiredState: string): boolean {
   return state === desiredState;
 }
 
+function isRenovateRestPr(ghPr: GhRestPr): boolean {
+  return (
+    config.forkMode ||
+    config.ignorePrAuthor ||
+    (ghPr?.user?.login && config?.renovateUsername
+      ? ghPr.user.login === config.renovateUsername
+      : true)
+  );
+}
+
 export async function getPrList(): Promise<Pr[]> {
-  logger.trace('getPrList()');
-  if (!config.prList) {
-    logger.debug('Retrieving PR list');
-    let prList: GhRestPr[];
+  if (!config.prCacheReady) {
     try {
-      prList = (
-        await githubApi.getJson<GhRestPr[]>(
-          `repos/${
-            config.parentRepo || config.repository
-          }/pulls?per_page=100&state=all`,
-          { paginate: true }
-        )
-      ).body;
+      let requestsTotal = 0;
+      let apiQuotaAffected = false;
+      let needNextPageFetch = true;
+      let needNextPageSync = true;
+
+      let pageIdx = 1;
+      while (needNextPageFetch && needNextPageSync) {
+        const repo = config.parentRepo || config.repository;
+        const urlPath = `repos/${repo}/pulls?per_page=100&state=all&sort=updated&direction=desc&page=${pageIdx}`;
+
+        const opts: GithubHttpOptions = { paginate: false };
+        if (pageIdx === 1) {
+          const etag = config.prCache.etag;
+          if (etag) {
+            opts.headers = { 'If-None-Match': etag };
+          } else {
+            // Speed up initial fetch
+            opts.paginate = true;
+          }
+        }
+
+        const res = await githubApi.getJson<GhRestPr[]>(urlPath, opts);
+        apiQuotaAffected = true;
+        requestsTotal += 1;
+
+        const nextPage = parseLinkHeader(res.headers?.link)?.next;
+        needNextPageFetch = !!nextPage;
+
+        if (pageIdx === 1) {
+          if (res.statusCode === 304) {
+            apiQuotaAffected = false;
+            break;
+          }
+
+          const etag = res.headers.etag;
+          if (etag) {
+            config.prCache.etag = etag;
```

Comment:
Small concern that you're writing the new etag before writing the new data, but it seems like nothing should go wrong in between ?

Reply:
yeah, i think we should update etag after the loop